src/WpfMath/UserDefinedColorParser.cs

```diff
@@ -0,0 +1,264 @@
+                        break;
+                    }
+                case ColorStringTypes.None:
+                    throw new TexParseException($"The text: {input} could not be parsed.");
+                default:
+                    break;
+            }
+            return resultColor;
+        }
+
+        #region Color Parsing Helpers
+        /// <summary>
+        /// Specifies the type of <see cref="Color"/> the string might represent.
+        /// </summary>
+        private enum ColorStringTypes
+        {
+            /// <summary>
+            /// Type is a <see cref="string"/> consisting of byte values of the color.
+            /// </summary>
+            Byte_longString,
+            /// <summary>
+            /// Type is a <see cref="string"/> consisting of byte values of the color.<para/>
+            /// Alpha channel is strictly = 255.
+            /// </summary>
+            Byte_shortString,
+            //FloatString,----->Will include this later.
+            /// <summary>
+            /// Type is a Hexadecimal <see cref="string"/> consisting of Hexadecimal values of the color.
+            /// </summary>
+            Hex_longString,
+            /// <summary>
+            /// Type is a Hexadecimal <see cref="string"/> consisting of Hexadecimal values of the color.<para/>
+            /// Alpha channel is strictly = FF.
+            /// </summary>
+            Hex_shortString,
+
+
+            /// <summary>
+            /// An unrecognized ColorStringType.
+            /// </summary>
+            None
+        }
+
+        /// <summary>
+        /// Gets the type of <see cref="ColorStringTypes"/> from the <paramref name="input"/>.
+        /// </summary>
+        /// <param name="input"></param>
+        /// <returns></returns>
+        private static ColorStringTypes GetColorString(string input)
+        {
+            if (IsByteHexTrain(input,6)==true)
+            {
+                return ColorStringTypes.Hex_shortString;
+            }
+            else if (IsByteHexTrain(input,8))
+            {
+                return ColorStringTypes.Hex_longString;
+            }
+            else if (IsByteTrain(input, 3) == true)
+            {
+                return ColorStringTypes.Byte_shortString;
+            }
+            else if (IsByteTrain(input, 4) == true)
+            {
+                return ColorStringTypes.Byte_longString;
+            }
+            else
+            {
+                return ColorStringTypes.None;
+            }
+        }
+
+        /// <summary>
+        /// Returns a value that tells if the <paramref name="input"/> contains byte values, separated by a ",".
+        /// </summary>
+        /// <example>
+        /// 23,78,56=>true
+        /// 789,fp=>false
+        /// </example>
+        /// <param name="input"></param>
+        /// <param name="num"></param>
+        /// <returns></returns>
+        private static bool IsByteTrain(string input, int num)
```

Comment:
1. This check is a little bit faulty. It seems like it will succeed e.g. for string `"0,0,quack,0,0"` for `num = 4`. Please fix that!
2. I think that this whole method could be replaced with a concise one-liner. Will you try to do it youself or do you need my help with that?